### PR TITLE
docs: refine patch release issue template

### LINKED ISSUE #3848: Further improve release issue templates
While releasing a few patches of Crossplane we noticed the following issue we should address in the current templates:
- [x] We should explicitly mention that crossplane/test has only to be updated if the patch is being cut for the latest patch release of the latest supported release [here](https://github.com/crossplane/crossplane/blob/master/.github/ISSUE_TEMPLATE/patch_release.md?plain=1#L33)
- [x] We should explicitly state how to announce on slack/twitter the new releases

### PR DESCRIPTION
**Pull request by phisco**
### Description of your changes

Fixes #3848 

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

N.A.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by phisco** [comment]
@jbw976 I'm not sure how to expand on the twitter step, any suggestion? Any specific channel/person the tweet should be requested to? I was going to write to ping you, but I'm not sure it's actually the right way, it's only the way I did it so far 😆

**Review by lsviben — APPROVED** [review]
Good updates based on our latest experience! Thanks @phisco

**Comment by jbw976** [comment]
> @jbw976 I'm not sure how to expand on the twitter step, any suggestion? 

We can say "Reach out to a Crossplane maintainer or steering committee member" and maybe link to https://github.com/crossplane/crossplane/blob/master/OWNERS.md

File: .github/ISSUE_TEMPLATE/patch_release.md [hunk 1]
```diff
@@ -30,9 +30,9 @@ examples of each step, assuming vX.Y.Z is being cut.
   - [ ] `xp/getting-started-with-gcp`
 - [ ] Run the [Promote workflow][promote-workflow] with channel `stable` on the `release-X.Y` branch and verified that the tagged build version exists on the [releases.crossplane.io] `stable` channel at `stable/vX.Y.Z/...`.
 - [ ] Published a [new release] for the tagged version, with the same name as the version and descriptive release notes, taking care of generating the changes list selecting as "Previous tag" `vX.Y.<Z-1>`, so the previous patch release for the same minor.
-- [ ] Updated the [`crossplane/test` repo test workflows][crossplane-test-workflows] to ensure the checkout release branch and helm install version(s) point at the new release across all the workflow files.
+- [ ] Only if cutting a patch for the latest supported release: updated the [`crossplane/test` repo test workflows][crossplane-test-workflows] to ensure the checkout release branch and helm install version(s) point at the new release across all the workflow files.
 - [ ] Ensured that users have been notified of the release on all communication channels:
-  - [ ] Slack
+  - [ ] Slack: `#announcements` channel on Crossplane's Slack workspace.
   - [ ] Twitter
```
Comment (jbw976) [inline .github/ISSUE_TEMPLATE/patch_release.md:36]: while we're here, do you want to go ahead and incorporate this twitter guidance: https://github.com/crossplane/crossplane/pull/3858#issuecomment-1478428286

**Review by jbw976 — COMMENTED** [review]
Thanks @phisco! A couple comments for you to look at 😉

**Comment by phisco** [comment]
@jbw976, totally missed your comment, sorry! I'll add that tomorrow 🙏

**Review by hasheddan — APPROVED** [review]

**Review by jbw976 — APPROVED** [review]
sweet, these updates look good @phisco! 💪

**Pull request merged**